packages/medusa/src/services/order.js

```diff
@@ -477,6 +477,9 @@ class OrderService extends BaseService {
               .withTransaction(manager)
               .cancelPayment(payment)
           }
+          await this.cartService_
+            .withTransaction(manager)
+            .update(cart.id, { payment_authorized_at: null })
```

Comment:
In case we cancel the payment, we should also cancel the payment authorization on the cart.